kourendlibrary/src/main/java/net/runelite/client/plugins/kourendlibrary/KourendLibraryCustomerOverlay.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018, dekvall <github.com/dekvall>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.kourendlibrary;
+
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import net.runelite.api.Client;
+import net.runelite.api.Item;
+import net.runelite.api.Player;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.components.PanelComponent;
+import net.runelite.client.ui.overlay.components.TitleComponent;
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+
+class KourendLibraryCustomerOverlay extends Overlay
+{
+	private static final int OVERLAY_WIDTH = 170;
+	private final Library library;
+	private final Client client;
+
+	private final KourendLibraryPlugin plugin;
+	private final PanelComponent panelComponent = new PanelComponent();
+	private final KourendLibraryConfig config;
+
+
+	@Inject
+	KourendLibraryCustomerOverlay(Library library, Client client, KourendLibraryPlugin plugin, KourendLibraryConfig config)
+	{
+		this.library = library;
+		this.client = client;
+		this.plugin = plugin;
+		this.config = config;
+		setPosition(OverlayPosition.TOP_LEFT);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		if (!config.showRequestedBookOverlay())
+		{
+			return null;
+		}
+
+		Player player = client.getLocalPlayer();
+		if (player == null)
+		{
+			return null;
+		}
+
+		WorldPoint playerLoc = player.getWorldLocation();
+
+		if (playerLoc.getRegionID() != KourendLibraryPlugin.REGION)
```

Comment:
Can you change this to only add the overlay to manager when entering the region? (could also check for the config; config cache only has 256 slots and even if cached unnecessary calls generate more garbage)